src/evo/mnhftx.cpp

```diff
@@ -229,10 +229,7 @@ std::optional<CMNHFManager::Signals> CMNHFManager::ProcessBlock(const CBlock& bl
             return signals;
         }
         for (const auto& versionBit : new_signals) {
-            if (Params().IsValidMNActivation(versionBit, pindex->GetMedianTimePast())) {
-                signals.insert({versionBit, mined_height});
-            }
-
+            signals.insert({versionBit, mined_height});
```

Comment:
3cd7a422ff2840bde73116593eb4c091aba6a2b9: how's this duplicated?

Reply:
see lines [222...225 ](https://github.com/dashpay/dash/pull/6225/files/6c6980832c8579c288988eadd6c25c739187c81a..3cd7a422ff2840bde73116593eb4c091aba6a2b9#diff-59f8e9f1b35c1428332caab753a818e3b2146e73bb6c998a2aed5f7eddbc6fa1R222-L225)

This condition already checked for each signal in new_signals above and if it fails for any signal std::nullopt will be a result of this function